### PR TITLE
fix(Router): Load attribute routes of all apps when not app is specified

### PR DESCRIPTION
**Pull request by provokateurin**
## Summary

Found while debugging https://github.com/nextcloud/server/issues/47479 (also necessary for the fix to it).

The problem was that when `loadRoutes(null)` was called the attribute routes should have been loaded for all apps since no app was specified. It only loaded the traditional routes.php files and then continued to set `loaded = true`. Any future call would return early, even if an app was specified because everything must be loaded already.
This logic now matches the loading process of routes.php which it should have done from the start.
Unfortunately back then I didn't find this problem and it seems to only trigger in specific places where the app is not known (e.g. when loading the `/index.php` path where the server has to load the routes first and then has to figure out which app should be used).

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: lib/private/Route/Router.php [hunk 1]
```diff
@@ -125,23 +129,10 @@ public function loadRoutes($app = null) {
...
-				$this->root->addCollection($collection);
 
-				// Also add the OCS collection
-				$collection = $this->getCollection($requestedApp . '.ocs');
-				$collection->addPrefix('/ocsapp');
-				$this->root->addCollection($collection);
-			}
+			$this->loadAttributeRoutes($app);
 		}
+		$this->eventLogger->start('route:load:' . $requestedApp, 'Loading Routes for ' . $requestedApp);
```
Comment (nickvergessen) [inline lib/private/Route/Router.php:135]: this is now not the start anymore?

File: lib/private/Route/Router.php [hunk 2]
```diff
@@ -413,6 +404,27 @@ protected function fixLegacyRootName(string $routeName): string {
 		return $routeName;
 	}
 
+	private function loadAttributeRoutes(string $app): void {
+		if (!in_array($app, \OC_App::getEnabledApps())) {
```
Comment (come-nc) [inline lib/private/Route/Router.php:408]:
```suggestion
		if (!$this->appManager->isEnabledForUser($app)) {
```

**Review by come-nc — APPROVED** [review]

**Review by artonge — APPROVED** [review]

**Pull request merged**